### PR TITLE
[depends] bump openssl to 1.1.0h

### PR DESCRIPTION
**Pull request by Rechi**
## Description
- bump openssl to version 1.1.0h
- build static libs only, instead of building shared and static and then delete the shared ones
- update CA trust store

## How Has This Been Tested?
compiled depends and kodi for android, ios and osx

## Types of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Cosmetic change (non-breaking change that doesn't touch code)
- [ ] None of the above (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by Rechi** [comment]
@wsnipex ping

File: tools/depends/target/openssl/Makefile [hunk 1]
```diff
@@ -3,23 +3,23 @@ DEPS= ../../Makefile.include Makefile
...
 # lib name, version
 LIBNAME=openssl
-VERSION=1.0.2l
+VERSION=1.1.0h
 SOURCE=$(LIBNAME)-$(VERSION)
 ARCHIVE=$(SOURCE).tar.gz
 
 # configuration settings
-CONFIGURE=MACHINE=$(PLATFORM) ./config shared zlib --openssldir=$(PREFIX) --with-zlib-include=$(PREFIX)/include --with-zlib-lib=$(PREFIX)/lib
+CONFIGURE=MACHINE=$(PLATFORM) ./config no-shared zlib no-asm --prefix=$(PREFIX) --with-zlib-include=$(PREFIX)/include --with-zlib-lib=$(PREFIX)/lib
```
Comment (wsnipex) [inline tools/depends/target/openssl/Makefile]: imho we shouldn't disable asm where it works. historically it was disabled on ios and osx only.

Comment (Rechi) [reply tools/depends/target/openssl/Makefile]: reason why I disabled it is the following build error

```/usr/bin/ld.gold: error: ../tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libcrypto.a(aesni-x86_64.o): multiple definition of 'aesni_encrypt'
/usr/bin/ld.gold: /home/jenkins/workspace/LINUX-64/tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libgnutls.a(aesni-x86_64.o): previous definition here
/usr/bin/ld.gold: error: ../tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libcrypto.a(aesni-x86_64.o): multiple definition of 'aesni_decrypt'
/usr/bin/ld.gold: /home/jenkins/workspace/LINUX-64/tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libgnutls.a(aesni-x86_64.o): previous definition here
/usr/bin/ld.gold: error: ../tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libcrypto.a(aesni-x86_64.o): multiple definition of 'aesni_ecb_encrypt'
/usr/bin/ld.gold: /home/jenkins/workspace/LINUX-64/tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libgnutls.a(aesni-x86_64.o): previous definition here
/usr/bin/ld.gold: error: ../tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libcrypto.a(aesni-x86_64.o): multiple definition of 'aesni_ccm64_encrypt_blocks'
/usr/bin/ld.gold: /home/jenkins/workspace/LINUX-64/tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libgnutls.a(aesni-x86_64.o): previous definition here
/usr/bin/ld.gold: error: ../tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libcrypto.a(aesni-x86_64.o): multiple definition of 'aesni_ccm64_decrypt_blocks'
/usr/bin/ld.gold: /home/jenkins/workspace/LINUX-64/tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libgnutls.a(aesni-x86_64.o): previous definition here
/usr/bin/ld.gold: error: ../tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libcrypto.a(aesni-x86_64.o): multiple definition of 'aesni_ctr32_encrypt_blocks'
/usr/bin/ld.gold: /home/jenkins/workspace/LINUX-64/tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libgnutls.a(aesni-x86_64.o): previous definition here
/usr/bin/ld.gold: error: ../tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libcrypto.a(aesni-x86_64.o): multiple definition of 'aesni_xts_encrypt'
/usr/bin/ld.gold: /home/jenkins/workspace/LINUX-64/tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libgnutls.a(aesni-x86_64.o): previous definition here
/usr/bin/ld.gold: error: ../tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libcrypto.a(aesni-x86_64.o): multiple definition of 'aesni_xts_decrypt'
/usr/bin/ld.gold: /home/jenkins/workspace/LINUX-64/tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libgnutls.a(aesni-x86_64.o): previous definition here
/usr/bin/ld.gold: error: ../tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libcrypto.a(aesni-x86_64.o): multiple definition of 'aesni_cbc_encrypt'
/usr/bin/ld.gold: /home/jenkins/workspace/LINUX-64/tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libgnutls.a(aesni-x86_64.o): previous definition here
/usr/bin/ld.gold: error: ../tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libcrypto.a(aesni-x86_64.o): multiple definition of 'aesni_set_decrypt_key'
/usr/bin/ld.gold: /home/jenkins/workspace/LINUX-64/tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libgnutls.a(aesni-x86_64.o): previous definition here
/usr/bin/ld.gold: error: ../tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libcrypto.a(aesni-x86_64.o): multiple definition of 'aesni_set_encrypt_key'
/usr/bin/ld.gold: /home/jenkins/workspace/LINUX-64/tools/depends/xbmc-depends/x86_64-linux-gnu-debug/lib/libgnutls.a(aesni-x86_64.o): previous definition here```

Comment (wsnipex) [reply tools/depends/target/openssl/Makefile]: meh..
ok, I guess it's not worth it to try to patch around it.

**Review by wsnipex — APPROVED** [review]

**Pull request merged**